asserts/ifacedecls_test.go

```diff
@@ -956,6 +985,79 @@ func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsDeviceScope
 	}
 }
 
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsSideArityConstraints(c *C) {
+	m, err := asserts.ParseHeaders([]byte(`iface:
+  allow-auto-connection: true`))
+	c.Assert(err, IsNil)
+
+	rule, err := asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+	c.Assert(err, IsNil)
+
+	// defaults
+	c.Check(rule.AllowAutoConnection[0].SlotsPerPlug, Equals, asserts.SideArityConstraint{N: 1})
+	c.Check(rule.AllowAutoConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	c.Check(rule.AllowConnection[0].SlotsPerPlug.Any(), Equals, true)
+	c.Check(rule.AllowConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	// allow-connection => *
```

Comment:
A larger comment, even if it is just a link to https://forum.snapcraft.io/t/plug-slot-declaration-rules-greedy-plugs/12438 would be helpful for future reviewers to understand the expectations this tests is verifying.